modin/core/io/column_stores/feather_dispatcher.py

```diff
@@ -52,13 +52,15 @@ def _read(cls, path, columns=None, **kwargs):
             import_optional_dependency(
                 "pyarrow", "pyarrow is required to read feather files."
             )
-            from pyarrow.feather import read_feather
+            from pyarrow import ipc
 
             with OpenFile(
                 path,
                 **(kwargs.get("storage_options", None) or {}),
             ) as file:
-                df = read_feather(file)
-            # pyarrow.feather.read_feather doesn't support columns as pandas.Index
-            columns = list(df.columns)
+                # Opens the file to extract its metadata
+                reader = ipc.open_file(file)
+            # TODO: pyarrow's schema contains much more metadata than just column names, it also
+            # has dtypes and index information that we could use when building a dataframe
+            columns = reader.schema.names
         return cls.build_query_compiler(path, columns, use_threads=False)
```

Comment:
I'm wondering why use_threads is False here...

Reply:
When using a distributed engine like Ray/Dask/Unidist we probably don't want to use threading in remote functions in order not to get into oversubscribing.